Model/Sync/Category/Processor/ProcessByCategory.php

```diff
@@ -140,27 +149,20 @@ public function processEntity($retryCategoryIds = [])
         $currentTime = date('Y-m-d H:i:s');
         $batchSize = $this->config->getConfig('product_sync_limit');
         $existColls = [];
-        $attributeId = $this->data->getAttributeId(Config::CATEGORY_SYNC_ATTR_CODE);
+        /** @var \Magento\Store\Model\Store  $currentStore**/
+        $currentStore = $this->storeManager->getStore();
+        $rootCategoryId = $currentStore->getRootCategoryId();
         $collection = $this->categoryCollectionFactory->create();
-        $collection->addAttributeToSelect('*');
-        $collection->addNameToResult();
-        if ($retryCategoryIds) {
-            $collection->getSelect()
-                ->where('e.entity_id in (?)', $retryCategoryIds);
-        }
-        $collection->getSelect()->joinLeft(
-            ['at' => $this->resourceConnection->getTableName('catalog_category_entity_int')],
-            'e.' . $this->entityIdFieldValue . ' = at.' . $this->entityIdFieldValue .
-            ' AND at.attribute_id = ' . $attributeId .
-            ' AND at.store_id=\'' . $this->config->getStoreId() . '\'',
-            null
-        );
+        $collection->addAttributeToFilter('path', ['like' => "1/{$rootCategoryId}/%"]);
```

Comment:
Can you please explain why do we need this?

Reply:
This is how it is filtering the categories for the current store view in the loop. Earlier it was actually collecting the categories which was not assigned to a particular store

Reply:
Got it.
Doesn't this mean that there's also a bug in the collection products sync?
As far as I understand, currently, we're syncing the relation of a product to **all** of its collection, not only the collections that are relevant store view.

Reply:
Hi @LiorGingi ,

Can you please suggest how should we fix it? Asking this because , from products page magento allows a product can be mapped to a category which doesn't belongs to that store. As a product can be assigned to multiple stores, how should we fix this ? Should we allow syncing the relation of a product to all of its collection OR should we restrict per store scope as we do it in processbycategory class?

OR should we remove the storewise restricion from processbycategory method?

Please suggest